src/components/PlantsPrimaryPage/PlantsPrimaryPageView.tsx

```diff
@@ -25,7 +25,7 @@ export type PlantsPrimaryPageViewProps = {
   actionButton?: ButtonProps;
   children: React.ReactNode; // primary content for this page
   isEmptyState?: boolean; // optional boolean to indicate this is an empty state view
-  onSelect: (plantingSite: PlantingSite) => void; // planting site selected, id of -1 refers to All
+  onSelect: (plantingSite: PlantingSite | undefined) => void; // planting site selected, id of -1 refers to All
   plantingSites: PlantingSite[] | undefined;
```

Comment:
I'm still not super great with typescript. What are the benefits of doing `plantingSite: PlantingSite | undefined` instead of just `plantingSite?: PlantingSite`?

Reply:
No benefit here, I guess. I was getting an error of undefined not being an option in that function and just added like that. But I will add it as `plantingSite?: PlantingSite` since that will also allow nullable (null) options